src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,58 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("The user doesn't exist");
+        }
+        validateLogin(user.getLogin());
+        validatePassword(user.getPassword());
+        validateAge(user.getAge());
+        return storageDao.add(user);
+    }
+
+    private void validateLogin(String login) {
+        if (login == null) {
+            throw new RegistrationException("Login can't be null.");
+        }
+        if (login.length() < MIN_LENGTH) {
+            throw new RegistrationException(String
+                    .format("Your login should contain %s MIN_LENGTH or more symbols.",
+                            MIN_LENGTH));
```

Comment:
minor 

you can use `formatted` method directly on a string instead of static `format`

Reply:
Thank you, very userful!